src/main/java/core/basesyntax/strategy/ShopServiceImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy;
+
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
+        for (FruitTransaction transaction : fruitTransactions) {
+            OperationHandler handler = operationStrategy
+                    .getOperationHandler(transaction.getOperation());
+            handler.handle(transaction);
```

Comment:
Consider adding a check to handle cases where `getOperationHandler` might return `null` if a handler is not found for a given operation. This will prevent potential `NullPointerException` and improve robustness.